### PR TITLE
Fix toolbar order (revisited)

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

The designer keeps on wrecking havoc with toolbars - i.e. the order appears to be semi random and not [expected](https://github.com/gitextensions/gitextensions/pull/8719#discussion_r559495096). 
Revisit the toolbar layout workaround, and add assertions to validate the outcome.

Related #8719

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by gerhardol — COMMENTED** [review]
+1

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -511,6 +500,35 @@ Brush UpdateCommitButtonAndGetBrush(IReadOnlyList<GitItemStatus> status, bool sh
...
+                }
+
+#if DEBUG
+                // 3. Assert the layout is expected
+                for (int i = toolStrips.Length - 1; i > 0; i--)
+                {
+                    // Assert all toolbars on the same row
+                    Debug.Assert(toolStrips[i].Top == 0, $"{toolStrips[i].Name} must be on placed on the 1st row");
+
+                    // Assert the correct order of toolbars
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormBrowse.cs:527]:
```suggestion
                // 3. Assert all toolbars on the same row
                foreach (ToolStrip toolStrip in toolStrips)
                {
                    Debug.Assert(toolStrip.Top == 0, $"{toolStrip.Name} must be placed on the 1st row");
                }

                // 4. Assert the correct order of toolbars
                for (int i = toolStrips.Length - 1; i > 0; i--)
                {
```

**Review by mstv — APPROVED** [review]
👍 Should work. Two nits in the assertions.

File: GitUI/CommandsDialogs/FormBrowse.cs (hunk 1 above)
Comment (RussKie) [reply GitUI/CommandsDialogs/FormBrowse.cs:527]: Why is it better?

Comment (mstv) [reply GitUI/CommandsDialogs/FormBrowse.cs:527]: Because it asserts *all* toolbars as the comment promises.

**Pull request merged**